core/include/sched_round_robin.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    Sched_RR
+ * @ingroup     sys
+ * @brief       This module module provides round robin scheduling within
+ *              for all runnable threads within each not masked priority.
+ *              Priority 0 is masked by default.
+ *              There is no fainess involved in this scheuler. It schedules,
+ *              when the timer ticks. Keep it as simple as possible.
+ *
+ * @note        start using start_schedule_rr()
+ *
+ * @{
+ *
+ * @file
+ * @brief       Round Robin Scheduler
+ *
+ * @author      Karl Fessel <karl.fessel@ovgu.de>
+ *
+ */
+
+#if !defined(SCHED_RR_TIMER) || defined(DOXYGEN)
+/**
+ * @brief Time between round robin calls in µs
+ *
+ * @details Defaults to 1ms
+ */
+#define SCHED_RR_TIMER 1000
+#endif
+
+#if !defined(SCHED_RR_MASK) || defined(DOXYGEN)
+/**
+ * @brief masks off priorities that should not be scheduled default: 0 is masked
+ */
+#define SCHED_RR_MASK 1<<0
+#endif
+
+/**
+ * @brief remove the round robin timer
+ */
+void remove_sched_round_robin(void);
+
+/**
+ * @brief setup the round robin timer for prio
+ */
+void set_sched_round_robin(uint8_t prio);
```

Comment:
Missing documentation for parameter `prio`.

It would be nice if all functions would contain a common prefix. E.g. `sched_rr_`...

Reply:
i will switch the function names around  from "set_sched_round_robin" to "sched_round_robin_set" 
since the is no need to keep set_.. public i will be static inline